### PR TITLE
Breathing Tube and Arm Toolset Augments

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Credit to TG for this.

Ports the breathing tube and arm toolset augments.

This is basically an augment that allows you to use internals without a breath mask of any sort; it additionally makes you immune to oxygen loss from being neck grabbed. If you get EMP'd, then it causes minor losebreath.

Can make it at R&D if you have materials 2 and biotech 3; you implant it in the mouth.

The arm toolset augment is an augment that gets implanted in either your left or right arm (use a screwdriver to change which arm it can be implanted in). The augment has a full cyborg toolset suite in it.

You will now also be able to produce IPC arm charging tools at mechfabs

Code wise:
- "chest arm" augments are gone; they're now full proper arm augments; you implant them in the arm and manipulate them in the arm; everything that was on the chest-arm system has been converted over.

:cl: Fox McCloud
add: Adds in breathing tube augment to R&D
add: Adds in arm toolset augment to R&D
tweak: IPC cell charging device is now inside their right arm, as opposed to chest
add: IPC cell charging device can now be produced in mech fabricators
/:cl:

### CODE REVIEW

**Comment by scrubmcnoob** [comment]
A chain smoker's dream right here.

On a more serious note, I'm really liking these augment PRs. Great moves.

**Comment by Ralta** [comment]
NO MORE VOX MASKS? YEEEEEES. +1

**Comment by Fox-McCloud** [comment]
Updated.

I went ahead and ported over TG's arm implanted augments; the only one accessible to players is the toolset augment; it's a full set of cyborg tools (twice as fast as normal use) that can be extended on use.

GO GO GADGET MULTITOOL!

Also IPC's charging organ now utilizes this new system; their charger is now in their right arm instead of their chest.

File: code/modules/research/designs/medical_designs.dm [hunk 1]
```diff
@@ -260,6 +260,28 @@
...
+/datum/design/cyberimp_breather
+	name = "Breathing Tube Implant"
+	desc = "This simple implant adds an internals connector to your back, allowing you to use internals without a mask and protecting you from being choked."
+	id = "ci-breather"
+	req_tech = list("materials" = 2, "biotech" = 3)
+	build_type = PROTOLATHE | MECHFAB
+	construction_time = 35
+	materials = list(MAT_METAL = 600, MAT_GLASS = 250)
+	build_path = /obj/item/organ/internal/cyberimp/mouth/breathing_tube
+	category = list("Misc", "Medical Designs")
```
Comment (Fox-McCloud) [inline code/modules/research/designs/medical_designs.dm]: Oh; this needs to be fixed.

File: code/modules/research/designs/medical_designs.dm [hunk 2]
```diff
@@ -260,6 +260,28 @@
...
+/datum/design/cyberimp_toolset
+	name = "Toolset Arm Implant"
+	desc = "A stripped-down version of engineering cyborg toolset, designed to be installed on subject's arm."
+	id = "ci-toolset"
+	req_tech = list("materials" = 3, "engineering" = 4, "biotech" = 4, "powerstorage" = 4)
+	build_type = PROTOLATHE | MECHFAB
+	materials = list (MAT_METAL = 2500, MAT_GLASS = 1500, MAT_SILVER = 1500)
+	construction_time = 200
+	build_path = /obj/item/organ/internal/cyberimp/arm/toolset
+	category = list("Misc", "Medical Designs")
```
Comment (Fox-McCloud) [inline code/modules/research/designs/medical_designs.dm]: Same.

File: code/modules/surgery/organs/organ_internal.dm [hunk 3]
```diff
@@ -5,8 +5,8 @@
 	force = 1
 	w_class = WEIGHT_CLASS_SMALL
 	throwforce = 0
-	var/zone = "chest"
```
Comment (Fox-McCloud) [inline code/modules/surgery/organs/organ_internal.dm]: This is completely unused.

File: code/modules/surgery/organs/augments_arms.dm [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+	if(istype(holder, /obj/item/device/flash/armimplant))
+		var/obj/item/device/flash/F = holder
+		F.set_light(7)
+
+	var/arm_slot = (parent_organ == "r_arm" ? slot_r_hand : slot_l_hand)
+	var/obj/item/arm_item = owner.get_item_by_slot(arm_slot)
+
+	if(arm_item)
+		if(!owner.unEquip(arm_item))
+			owner << "<span class='warning'>Your [arm_item] interferes with [src]!</span>"
```
Comment (Fox-McCloud) [inline code/modules/surgery/organs/augments_arms.dm]: `to_chat`

File: code/modules/surgery/organs/augments_arms.dm [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+			return
+		else
+			owner << "<span class='notice'>You drop [arm_item] to activate [src]!</span>"
```
Comment (Fox-McCloud) [inline code/modules/surgery/organs/augments_arms.dm]: `to_chat`

File: code/modules/surgery/organs/augments_arms.dm [hunk 6]
```diff
@@ -0,0 +1,315 @@
...
+
+	if(parent_organ == "r_arm" ? !owner.put_in_r_hand(holder) : !owner.put_in_l_hand(holder))
+		owner << "<span class='warning'>Your [src] fails to activate!</span>"
```
Comment (Fox-McCloud) [inline code/modules/surgery/organs/augments_arms.dm]: `to_chat`

**Comment by KingPhilipIII** [comment]
Whenever I give people tool arms on TG I would put one in each arm for good measure.

**Comment by taukausanake** [comment]
Think we can get some basic tools too? I personally would like a health and plant scanner in my arm just because. Notice the surgical tools don't have one. And maybe a botanist arm implant too with their tools.

A PDA arm implant would be pretty swanky. Pipboy level of nice. I wonder what that could be called...

For people with batteries for hearts does that mean they can or could recharge their heart by use of a power adapter implant? Or do battery hearts even have to do that? Would be cool and necessary I think

**Comment by granodd** [comment]
Can't say I'm really a fan of moving the IPCs charging implant into their arm rather than keeping it in their chest. Reasoning being that the chest is something that can't pop off, while an IPC's limbs don't want to get too cozy due to the danger of falling off as soon as a stiff breeze (read: antags or greytide) blow through the station. I also have to wonder if the implant would actually stay in the arm, or if it would drop out onto the ground along with the arm. Very painful if the latter, as that means the limb needs to be popped back on, and then have surgery done onto it to stick the implant back in.

It might not seem like much to move it into the arm, but it's still a decrease in the QoL for recharging as an IPC, more so than the bacon-strip of charging's first implementation was (due to the fact that before that they could just recharge even after EMP without having to worry that their bacon-strip might be damaged).

Otherwise, tools seem okay. A little wary of how low the breathing tube's tech is set, but I guess we'll see.

**Comment by Fox-McCloud** [comment]
The augment stays in the arm, even when it pops off (I specifically tested this), and it comes back when the arm is re-attached (again, tested).

**Comment by taukausanake** [comment]
@Fox-McCloud I think I've done it before, with slight success but I can't recall. I know I kalima'd someone in Robotics to transplant a cell. So it doesn't use power as a heart organ?

**Comment by alexkar598** [comment]
As a surgeon in the game,I did 2 heart transplant, 1 died the other nearly died.
I don't recommend making a heart transplant in any ways unsless you have abuctor surgery table

**Comment by Ralta** [comment]
@alexkar598 You can safely do heart transplants quite easily, so long as the actual patient isn't otherwise injured. Just gotta be sure to have a defib nearby.

But yeah, surgery tools would be badass. I love getting the IMS from RnD, but it's rare that they have the tech (or staff) to make it; getting a hemostat and retractor in my arms would be awesome!

**Comment by Rumiluntti** [comment]
Wait, full set fo tools that can't be removed from the greytider unless you chop the arm off?

**Comment by taukausanake** [comment]
Oh oh! Make defib implants that require both your hands to use! Might be a little outside this PR but one can dream.

I would love if you could emag those and have UNLIMITED POWER but that's maybe best for something else or just wizard

**Comment by Fox-McCloud** [comment]
it was on the 8th day, the maintainers said "let there be merge conflicts".

And there was much screaming and gnashing of teeth across the land.

**Pull request merged**